frontend/src/lib/components/apps/editor/settingsPanel/inputEditor/StaticInputEditor.svelte

```diff
@@ -24,17 +24,49 @@
 	import EditableSchemaDrawer from '$lib/components/schema/EditableSchemaDrawer.svelte'
 	import AppPicker from '$lib/components/wizards/AppPicker.svelte'
 
-	export let componentInput: StaticInput<any> | undefined
+	export let componentInput: (StaticInput<any> & { isLabeled?: boolean }) | undefined
 	export let fieldType: InputType | undefined = undefined
 	export let subFieldType: InputType | undefined = undefined
 	export let selectOptions: StaticOptions['selectOptions'] | undefined = undefined
 	export let placeholder: string | undefined = undefined
 	export let format: string | undefined = undefined
 	export let id: string | undefined
+	export let hasLabeledMode: boolean = false
 
 	const { onchange } = getContext<AppViewerContext>('AppViewerContext')
 
 	$: componentInput && onchange?.()
+
+	onMount(() => {
+		/** preserve the `labeled` toggling when switching from eval to static */
+		if (componentInput !== undefined) {
+			handleComponentInputChange(componentInput)
+		}
+	})
+	/**
+	 * Every components: switching from `static` to `eval` preserves the user data
+	 *
+	 * Multiselect: toggling `labeled` needs to mutate input value and input fieldType.
+	 * */
+	function handleComponentInputChange(newComponentInput: StaticInput<any>) {
+		componentInput = { ...componentInput, ...newComponentInput }
```

Comment:
that doesn't make sense, both are from componentInput

Reply:
When removing this line `componentInput = { ...componentInput, ...newComponentInput }`,
The component doesn't update correctly, it relates to Svelte's reactivity being handled by dependency variables mutation checks.

According to [sveltesociety.dev](https://www.sveltesociety.dev/recipes/svelte-language-fundamentals/reactivity)
> Svelte only cares about which variables are being reassigned, not the values to which those variables refer. [...] This is the sort of problem you may run into when dealing with objects. Since objects are passed by reference and not value

I tried using the solution  `componentInput = componentInput` at the top or bottom of the function without success (as suggested [there](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Svelte_reactivity_lifecycle_accessibility#reactivity_gotchas_updating_objects_and_arrays) and [there)](url)